File: rulepacks/2026.08.3/testdata/seguridad-ext.cs
// Fixture de semgrep --test — no es código del producto.
// Casos positivos (ruleid) y negativos (ok) para las reglas csharp-* y
// cors-wildcard-con-credenciales-net de seguridad-ext.yaml.
// Solo necesita parsear: los tipos no se resuelven.
namespace Fixtures
{
    public class SeguridadExtFixture
    {
        void SqlConcat(string idUsuario, SqlConnection conexion)
        {
            // ruleid: csharp-sql-concat
            var comando = new SqlCommand("SELECT id, nombre FROM usuarios WHERE id = " + idUsuario, conexion);
            // ruleid: csharp-sql-concat
            var interpolado = new NpgsqlCommand($"SELECT id, nombre FROM usuarios WHERE nombre = {idUsuario}", conexion);
            // ok: csharp-sql-concat
            var seguro = new SqlCommand("SELECT id, nombre FROM usuarios WHERE id = @id", conexion);
        }

        void Comandos(string entrada)
        {
            // ruleid: csharp-command-injection
            var infoInsegura = new ProcessStartInfo { FileName = "cmd.exe", UseShellExecute = true };
            // ruleid: csharp-command-injection
            Process.Start("cmd.exe", $"/c {entrada}");
            // ok: csharp-command-injection
            var infoSegura = new ProcessStartInfo { FileName = "git", UseShellExecute = false };
        }

        void JwtSinVerificar(JwtSecurityTokenHandler manejador, string tokenCrudo)
        {
            // ruleid: csharp-jwt-no-verify
            var parametros = new TokenValidationParameters { ValidateIssuerSigningKey = false };
            // ruleid: csharp-jwt-no-verify
            var contenido = manejador.ReadJwtToken(tokenCrudo);
            // ok: csharp-jwt-no-verify
            var parametrosSeguros = new TokenValidationParameters { ValidateIssuerSigningKey = true, ValidateLifetime = true };
        }

        void Evaluar(string codigo)
        {
            // ruleid: csharp-dynamic-eval
            CSharpScript.EvaluateAsync(codigo);
            // ok: csharp-dynamic-eval
            var numero = int.Parse(codigo);
        }

        void Criptografia(Aes cifrador)
        {
            // ruleid: csharp-crypto-debil
            var resumenInseguro = MD5.Create();
            // ruleid: csharp-crypto-debil
            cifrador.Mode = CipherMode.ECB;
            // ok: csharp-crypto-debil
            var resumenSeguro = SHA256.Create();
            // ok: csharp-crypto-debil
            cifrador.Mode = CipherMode.CBC;
        }

        void Aleatorios()
        {
            // ruleid: csharp-random-inseguro
            var generador = new Random();
            // ruleid: csharp-random-inseguro
            var valor = Random.Shared.Next();
            // ok: csharp-random-inseguro
            var seguro = RandomNumberGenerator.GetInt32(100);
        }

        void TlsPermisivo()
        {
            // ruleid: csharp-tls-verify-off
            ServicePointManager.ServerCertificateValidationCallback = (emisor, certificado, cadena, errores) => true;
            // ruleid: csharp-tls-verify-off
            var manejadorInseguro = new HttpClientHandler { ServerCertificateCustomValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator };
            // ok: csharp-tls-verify-off
            var manejadorSeguro = new HttpClientHandler { AllowAutoRedirect = false };
        }

        void Xml(XmlDocument documento)
        {
            // ruleid: csharp-xxe
            var opcionesInseguras = new XmlReaderSettings { DtdProcessing = DtdProcessing.Parse };
            // ruleid: csharp-xxe
            documento.XmlResolver = new XmlUrlResolver();
            // ok: csharp-xxe
            var opcionesSeguras = new XmlReaderSettings { DtdProcessing = DtdProcessing.Prohibit };
        }

        void Cors(CorsPolicyBuilder politica)
        {
            // ruleid: cors-wildcard-con-credenciales-net
            politica.AllowAnyOrigin().AllowCredentials();
            // ok: cors-wildcard-con-credenciales-net
            politica.WithOrigins("https://app.ejemplo.com").AllowCredentials();
        }
    }
}

File: rulepacks/2026.08.3/testdata/seguridad-ext.java
// Fixture de semgrep --test — no es código del producto.
// Casos positivos (ruleid) y negativos (ok) para las reglas java-* y
// cors-wildcard-con-credenciales-java de seguridad-ext.yaml.
// Solo necesita parsear: los tipos no se resuelven.
package fixtures;

import java.util.Arrays;

class SeguridadExtFixture {

    void sqlConcat(Statement stmt, Connection conn, String idUsuario) throws Exception {
        // ruleid: java-sql-concat
        stmt.executeQuery("SELECT id, nombre FROM usuarios WHERE id = " + idUsuario);
        // ok: java-sql-concat
        conn.prepareStatement("SELECT id, nombre FROM usuarios WHERE id = ?");
    }

    void comandos(String host, String comando) throws Exception {
        // ruleid: java-command-injection
        Runtime.getRuntime().exec("ping -c 1 " + host);
        // ruleid: java-command-injection
        new ProcessBuilder("sh", "-c", comando);
        // ok: java-command-injection
        ProcessBuilder listaSegura = new ProcessBuilder("ls", "-la");
    }

    Object deserializar(ObjectInputStream flujo) throws Exception {
        // ruleid: java-unsafe-deserialization
        Object objeto = flujo.readObject();
        // ruleid: java-unsafe-deserialization
        Yaml cargadorInseguro = new Yaml();
        // ok: java-unsafe-deserialization
        Yaml cargadorSeguro = new Yaml(new SafeConstructor());
        return objeto;
    }

    void jwtSinVerificar(JwtParser analizador, String tokenCrudo) {
        // ruleid: java-jwt-no-verify
        JWT.decode(tokenCrudo);
        // ruleid: java-jwt-no-verify
        analizador.parseClaimsJwt(tokenCrudo);
        // ok: java-jwt-no-verify
        analizador.parseClaimsJws(tokenCrudo);
    }

    void evaluar(ScriptEngine motor, String expresion) throws Exception {
        // ruleid: java-dynamic-eval
        motor.eval(expresion);
        // ok: java-dynamic-eval
        Integer.parseInt(expresion);
    }

    void criptografia() throws Exception {
        // ruleid: java-crypto-debil
        MessageDigest.getInstance("MD5");
        // ruleid: java-crypto-debil
        Cipher.getInstance("AES/ECB/PKCS5Padding");
        // ok: java-crypto-debil
        MessageDigest.getInstance("SHA-256");
        // ok: java-crypto-debil
        Cipher.getInstance("AES/GCM/NoPadding");
    }

    void aleatorios() {
        // ruleid: java-random-inseguro
        Random generador = new Random();
        // ok: java-random-inseguro
        SecureRandom seguro = new SecureRandom();
    }

    void tlsPermisivo(HttpsURLConnection conexion, SSLSocketFactory fabrica) {
        // ruleid: java-tls-verify-off
        conexion.setHostnameVerifier(NoopHostnameVerifier.INSTANCE);
        // ruleid: java-tls-verify-off
        TrustManager confiadoTodo = new X509TrustManager() {
            public void comprobar() { }
        };
        // ok: java-tls-verify-off
        conexion.setSSLSocketFactory(fabrica);
    }

    void xmlInseguro() throws Exception {
        // ruleid: java-xxe
        DocumentBuilderFactory fabricaInsegura = DocumentBuilderFactory.newInstance();
    }

    void xmlSeguro(DocumentBuilderFactory fabrica) throws Exception {
        fabrica.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
        // ok: java-xxe
        DocumentBuilderFactory otraFabrica = DocumentBuilderFactory.newInstance();
    }

    void extraer(String directorio, ZipEntry entrada) {
        // ruleid: java-zip-slip
        File destino = new File(directorio, entrada.getName());
        // ok: java-zip-slip
        File fijo = new File(directorio, "salida.txt");
    }

    void responder(HttpServletRequest req, HttpServletResponse resp) throws Exception {
        // ruleid: java-xss-response
        resp.getWriter().println(req.getParameter("nombre"));
        // ok: java-xss-response
        resp.getWriter().println("hola");
    }

    void cors(CorsConfiguration configuracion) {
        // ruleid: cors-wildcard-con-credenciales-java
        configuracion.setAllowedOrigins(Arrays.asList("*"));
        // ok: cors-wildcard-con-credenciales-java
        configuracion.setAllowedOrigins(Arrays.asList("https://app.ejemplo.com"));
    }
}
